src/utils/request.ts

```diff
@@ -30,12 +30,15 @@ export function parseUrl(request: Request): URL | undefined {
 }
 
 export function parseConditionalHeaders(headers: Headers): ConditionalHeaders {
-  const ifModifiedSince = headers.has('if-modified-since')
+  let ifModifiedSince = headers.has('if-modified-since')
     ? new Date(headers.get('if-modified-since')!)
     : undefined;
 
   if (ifModifiedSince instanceof Date) {
     ifModifiedSince.setSeconds(ifModifiedSince.getSeconds() + 1);
+  } else {
+    // Invalid date
```

Comment:
```suggestion
```
Nit: with the suggested method call being `isValidDate` I imagine this comment is redundant now